packages/react/src/components/Slider/Slider.tsx

```diff
@@ -478,9 +628,29 @@ export default class Slider extends PureComponent<SliderProps> {
       this.element?.ownerDocument.addEventListener(element, this.onDrag);
     });
 
+    const clientX = this.getClientXFromEvent(evt);
+
+    let activeHandle;
+    if (this.props.twoHandles) {
+      const distanceToLower = this.calcDistanceToHandle(
+        HandlePosition.LOWER,
+        clientX
+      );
+      const distanceToUpper = this.calcDistanceToHandle(
+        HandlePosition.UPPER,
+        clientX
+      );
+      if (distanceToLower <= distanceToUpper) {
+        activeHandle = HandlePosition.LOWER;
+      } else {
+        activeHandle = HandlePosition.UPPER;
+      }
```

Comment:
Could this be the culprit for the unexpectedly active thumb? Reset to whatever the current active handle is when `distanceToLower === distanceToUpper`?

Reply:
Nice catch. Turns out, you were close, and this was really helpful to quickly resolve it. The issue was that when the handles are pointing at the same value, `this.state.lowerLeft === this.state.lowerRight`, which meant that the `calcDistanceToHandle` always returned exactly the same value for both handles. In https://github.com/carbon-design-system/carbon/pull/14297/commits/3b3a7a065f7ce19d9f0102c5433211d0f64929d7 I offset the lower handle by it's width which more accurately places it for the purposes of the calculation, and we get the result we want. At least is was working in my testing locally 🤞 .